pymatviz/ptable.py

```diff
@@ -160,6 +162,231 @@ def count_elements(
+
+    # Turn off axis of subplots on the grid that don't correspond to elements
+    ax: plt.Axes
+    for ax in axes.flat:
+        ax.axis("off")
+
+    symbol_kwargs = symbol_kwargs or {}
+    symbol_kwargs.setdefault("fontsize", 18)
+
+    elem_class_colors = ELEM_CLASS_COLORS | (
+        color_elem_types if isinstance(color_elem_types, dict) else {}
+    )
+
+    # Get colormap
+    cmap = plt.get_cmap(colormap)
+
+    # Get the min/max values for norm calculation
+    all_values = [val for pair in data.values() for val in pair]
+    vmin = min(all_values)
+    vmax = max(all_values)
+
+    norm = Normalize(vmin=vmin, vmax=vmax)
+
+    # Scale data for easier color mapping
+    scaled_data = {key: [norm(val) for val in pair] for key, pair in data.items()}
+
+    for element in Element:
+        symbol = element.symbol
+        row, group = df_ptable.loc[symbol, ["row", "column"]]
+
+        ax = axes[row - 1][group - 1]
+        plot_data = np.array(data.get(symbol, []))
+        if len(plot_data) == 0 and on_empty == "hide":
+            continue
+
+        if color_elem_types:
+            elem_class = df_ptable.loc[symbol, "type"]
+            if color_elem_types in {"symbol", "both"}:
+                symbol_kwargs["color"] = elem_class_colors.get(elem_class, "black")
+            if color_elem_types in {"background", "both"}:
+                bg_color = elem_class_colors.get(elem_class, "white")
+                ax.set_facecolor((*mpl.colors.to_rgb(bg_color), 0.07))
+
+        ax.text(
+            *symbol_pos,
+            symbol_text(element)
+            if callable(symbol_text)
+            else symbol_text.format(elem=element),
+            ha="center",
+            va="center",
+            transform=ax.transAxes,
+            **symbol_kwargs,
+        )
+
+        ax.axis("on")
+
+        if anno_kwds:
+            defaults = dict(
+                text=lambda plot_vals: si_fmt_int(len(plot_vals)),
+                xy=(0.8, 0.8),
+                xycoords="axes fraction",
+                fontsize=8,
+                horizontalalignment="center",
+                verticalalignment="center",
+            )
+            if callable(anno_kwds):
+                annotation = anno_kwds(plot_data)
+            else:
+                annotation = anno_kwds
+                anno_text = anno_kwds.get("text")
+                if isinstance(anno_text, dict):
+                    anno_text = anno_text.get(symbol)
+                elif callable(anno_text):
+                    anno_text = anno_text(plot_data)
+                annotation["text"] = anno_text
+            ax.annotate(**(defaults | annotation))
+
+        if plot_data is not None:
+            # Fill areas above and below the line
+            x = np.linspace(0, 10, 10)
+            y = np.linspace(10, 0, 10)
+            ax.fill_between(x, y, 10, color=cmap(scaled_data[symbol][0]))
+            ax.fill_between(x, y, 0, color=cmap(scaled_data[symbol][1]))
```

Comment:
this is smart! 👍 i like this way of making the triangles. on the other hand, it could be finicky to support more than 2 triangles with this like [in this example](https://stackoverflow.com/a/66050636)?
![O3Pbi](https://github.com/janosh/pymatviz/assets/30958850/9d7ad1ed-a4e3-410c-a623-58e7a13530ba)

Reply:
Yes the 4-split method  looks very geeky. But it's certainly good to have the function. I believe we could do better with our line-split approach. 

I would have a try and implement a helper function to split a rectangle into even parts tomorrow.

Reply:
I'm playing around with a more clean yet powerful (helpfully) method: build a pie chart with the `matplotlib` standard `pie` API, and crop a rectangle out of it. In this way we could have any number of splits and don't need to hard code the separation lines, but I'm still playing around with the following toy code before integrating it into the `ptable` plotter:

```python
"""Toy code to split a rectangle into equal parts."""


import matplotlib.pyplot as plt
import numpy as np
from matplotlib.colors import Colormap, Normalize


def plot_split_rectangle(
    colors: list,
    start_phase: float = 0,  # TODO: control rotation
    ) -> None:
    # Plot the pie chart
    fig, ax = plt.subplots(figsize=(6, 6))
    ax.pie(np.ones(len(colors)), colors=colors)

    # Crop a central rectangle from the pie chart
    path = plt.Rectangle((-0.5, -0.5), 1, 1, fc='none', ec='none')
    ax.add_patch(path)

    ax.set_xlim(-0.5, 0.5)
    ax.set_ylim(-0.5, 0.5)

    # Hide axes spines
    ax.spines['top'].set_visible(False)
    ax.spines['right'].set_visible(False)
    ax.spines['bottom'].set_visible(False)
    ax.spines['left'].set_visible(False)

    # Hide axes ticks
    ax.tick_params(axis='both', which='both', bottom=False, top=False, left=False, right=False)

    ax.set_aspect('equal', adjustable='box')  # not seem to work?

    plt.show()

if __name__ == "__main__":
    # Prepare colors from Colormap
    data = [0, 10, 20, 30, 40, 60]

    cmap: Colormap = plt.get_cmap("coolwarm")

    # Normalize data
    norm = Normalize(vmin=min(data), vmax=max(data))
    normalized_data = [norm(datum) for datum in data]

    colors = [cmap(data) for data in normalized_data]

    # Test plotter
    plot_split_rectangle(colors)

```

Which gives: 
<img width="400" alt="image" src="https://github.com/janosh/pymatviz/assets/80093591/7b261bd3-0d22-422e-a0e4-899645a02cda">